### ISSUE TITLE
Memory use

### ISSUE THREAD

**Issue by yifliu3** [yyfz/Pi3#30]
Dear devs,

Thanks for opensourceing such great work! I'm curious about the memory use to train pi3 for each single gpu, if the bs is set to 1.

**Comment by yyfz** [comment]
Hi, [yifliu3](https://github.com/yifliu3)!

If the batch size is 1, with each sample containing 64 images at 224x224 resolution, the GPU memory usage should not exceed 35GB when using gradient checkpointing. You can reduce memory consumption by decreasing the number of images per batch.
